### ISSUE TITLE
`localized` command should validate input file before running

### ISSUE THREAD

**Issue by qtehfw** [jayrave/android_localization#9]
<img width="619" alt="Screen Shot 2020-02-20 at 12 52 28 PM" src="https://user-images.githubusercontent.com/59625999/74893229-fadfba00-53df-11ea-91dc-36e45fdee4de.png">

When I ran it today it ran into the above issue, but it had already started modifying the `strings.xml` files so `./app/src/main/res/values-ko/strings.xml` ended up in an invalid state:

<img width="931" alt="Screen Shot 2020-02-20 at 12 54 57 PM" src="https://user-images.githubusercontent.com/59625999/74893320-38444780-53e0-11ea-9617-5a91fd43cc68.png">

**Comment by qtehfw** [comment]
It could also search for unescaped apostrophes, and potentially escape them, before generating the outputs.

**Comment by jayrave** [comment]
Can you paste the line that led to this issue? The one from the CSV file

**Comment by qtehfw** [comment]
모델\"%1$s\"에 오류가 있습니다. 재업로드를 위해서는 <![DATA[<a href=\"https://app.fieldwire.com\">app.fieldwire.com</a>]]>에서 온라인에 접속하여 다시 시도해 주세요

**Comment by qtehfw** [comment]
I just had to add a `'C'` to the `<![DATA` and re-run the command

**Comment by jayrave** [comment]
Looks like even `localize` would fail if the default strings file has a string like this. I guess for this we can ask `localize` to perform validate before it starts its own run. 

For `localized`, it could take some more work since essentially we want to validate the CSV before the XML files are updated. May be we read in all the CSV strings, run validations in memory & proceed with XML updates only if all of them succeed?

**Comment by qtehfw** [comment]
Yeah, but I think in this case it was probably due to a typo in the translated string.

For `validate`, is the XML parsed into some kind of in-memory data structure before validating? If yes, does `localized` parse the csv into that same data structure?

**Comment by jayrave** [comment]
Sorry, was a bit busy with work!

Yes. We parse into `AndroidString` & then perform the validations. Localized also deals with strings in this format. We should be able to reuse stuff here.

Another idea: I guess these failures happen if the XML is mis-formatted like `<![DATA` instead of `<![CDATA`. To makes things easy for us, before `localize` & `localized` (may be even `validate`), we can just validate if strings are valid XML values
